h5py/_hl/dataset.py

```diff
@@ -213,10 +214,16 @@ def __len__(self):
         """
         return len(self._dset)
 
-    def __array__(self, dtype=None):
+    def __array__(self, dtype=None, copy=True):
+        if copy not in (True, _NUMPY_COPY_IF_NEEDED):
```

Comment:
The docs seem to suggest that NumPy 1.x never passes the `copy` parameter to `__array__` (e.g. [here](https://numpy.org/doc/stable/reference/arrays.classes.html#numpy.class.__array__) & [here](https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_FromArrayAttr) only document it taking a dtype argument). This fits with a brief test I did on NumPy 1.26.2. If that's right, then I think we can simplify this:

```suggestion
    def __array__(self, dtype=None, copy=True):
        if copy is False:
```

Reply:
good call. Indeed the `copy` argument in `__array__` is new in numpy 2.0 !